pkg/kubefedctl/orphaning/enable.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package orphaning
+
+import (
+	"io"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+
+	"sigs.k8s.io/kubefed/pkg/controller/sync"
+	"sigs.k8s.io/kubefed/pkg/kubefedctl/util"
+)
+
+var (
+	orphaning_enable_long = `
+		Prevents removal of managed resources from member clusters, when their federated 
+		resource is removed. Adds 'kubefed.k8s.io/orphan: true' as an annotation to the 
+		federated resource.
+
+		Current context is assumed to be a Kubernetes cluster hosting
+		the kubefed control plane. Please use the
+		--host-cluster-context flag otherwise.`
+
+	orphan_enable_example = `
+		# Enable the orphaning mode for federated resources with type FederatedDeployment and named foo 
+		kubefedctl orphaning enable FederatedDeployment foo --host-cluster-context=cluster1`
+)
+
+// newCmdEnableOrphaning adds 'kubefed.k8s.io/orphan: true' as an annotation to the given federated resource
+func newCmdEnableOrphaning(cmdOut io.Writer, config util.FedConfig) *cobra.Command {
+	opts := &orphanResource{}
+	cmd := &cobra.Command{
+		Use:     "enable <resource type> <resource name>",
+		Short:   "Enable to remain managed resources after removing the federated resource ",
```

Comment:
-> Enable the orphaning (i.e. retention) of resources managed by a federated resource upon its removal.

Reply:
done